buildSrc/src/main/java/org/elasticsearch/gradle/util/GradleUtils.java

```diff
@@ -139,6 +143,8 @@ public static void maybeConfigure(TaskContainer tasks, String name, Action<? sup
 
         setupIdeForTestSourceSet(project, testSourceSet);
 
+        disableTransitiveDependenciesForSourceSet(project, testSourceSet);
```

Comment:
I wonder if we could make this simpler by having something like `SourceSetContainer#all { disable Transitive reps }` in our `ElasticsearchJavaPlugin` instead of having calling this scattered across different plugins

Reply:
Indeed, that seems like a much better solution. Make it so :wink:

Reply:
🤡

Reply:
https://github.com/elastic/elasticsearch/pull/68095